### PR TITLE
PubMatic Bid Adapter : add GPP support

### PR DESCRIPTION
**Pull request by pm-nitin-shirsat**
## Type of change

- [ ] Bugfix
- [x] Feature
- [ ] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes

- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change

GPP support in PubMatic bid adapter

### CODE REVIEW

**Comment by bretg** [comment]
docs PR https://github.com/prebid/prebid.github.io/pull/4691

File: modules/pubmaticBidAdapter.js [hunk 1]
```diff
@@ -1366,6 +1375,12 @@ export const spec = {
       syncurl += '&us_privacy=' + encodeURIComponent(uspConsent);
     }
 
+    // GPP Consent
+    if (gppConsent?.gppString && gppConsent?.applicableSections?.length) {
+      syncurl += '&gpp=' + encodeURIComponent(gppConsent.gppString);
+      syncurl += '&gpp_sid=' + encodeURIComponent(gppConsent?.applicableSections);
```
[the remark this answers is not in the document]
Comment (pm-nitin-shirsat) [reply modules/pubmaticBidAdapter.js]: Thank you, Yes, it should encode as string, fixed it.

**Comment by pm-nitin-shirsat** [comment]
@patmmccann - As you mentioned, gpp_sid I had encoded as a string, can you please review it? Thank you!

**Comment by pm-nitin-shirsat** [comment]
@ChrisHuie or @patmmccann  - If anything is pending from my side to merge this PR do let me know? Thanks!

**Comment by patmmccann** [comment]
@pm-nitin-shirsat please feel free to commit to 7.54.x as well if you desire

**Pull request merged**